am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -165,7 +170,13 @@ public void revokeResourceAccess(@NotNull @Valid ExplicitAccessMetadata accessMe
         + "and permissions {{result.access.permissions}}")
     public FilteredResourceEnvelope filterResource(@NotBlank String userId,
                                                    @NotEmpty Set<@NotBlank String> userRoles,
-                                                   @NotNull @Valid Resource resource) {
+                                                   @NotNull @Valid Resource resource,
+                                                   @NotEmpty @Valid Map<JsonPointer, SecurityClassification>
+                                                       attributeSecurityClassification) {
+
+        if (attributeSecurityClassification.get(JsonPointer.valueOf("")) == null) {
+            throw new NoSuchElementException();
```

Comment:
could add message to give more useful exception

Reply:
Updated the exception to include the following message.
throw new NoSuchElementException("Root element not found in input Security Classification");